From 14782848ae647cf21ae8b3058c6ece8fbfe585a0 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Thu, 7 Dec 2017 15:29:46 +0200
Subject: [PATCH 051/767] arm64: dts: add dma coherent flags for DPAA 1.x
 Ethernet nodes

The performance is impacted if the memory is mapped as non coherent.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts | 1 +
 arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts | 1 +
 arch/arm64/boot/dts/freescale/qoriq-dpaa-eth.dtsi     | 7 +++++++
 3 files changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
index 0e7e7a2e4b7c..2f93c2668513 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
@@ -68,6 +68,7 @@
 	ethernet@9 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet7>;
+		dma-coherent;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
index bfe2f36c3030..87f598c59794 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-sdk.dts
@@ -68,6 +68,7 @@
 	ethernet@9 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet7>;
+		dma-coherent;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/qoriq-dpaa-eth.dtsi b/arch/arm64/boot/dts/freescale/qoriq-dpaa-eth.dtsi
index eb5af912b988..5c6be929f6de 100644
--- a/arch/arm64/boot/dts/freescale/qoriq-dpaa-eth.dtsi
+++ b/arch/arm64/boot/dts/freescale/qoriq-dpaa-eth.dtsi
@@ -37,30 +37,37 @@ fsldpaa: fsl,dpaa {
 	ethernet@0 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet0>;
+		dma-coherent;
 	};
 	ethernet@1 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet1>;
+		dma-coherent;
 	};
 	ethernet@2 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet2>;
+		dma-coherent;
 	};
 	ethernet@3 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet3>;
+		dma-coherent;
 	};
 	ethernet@4 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet4>;
+		dma-coherent;
 	};
 	ethernet@5 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet5>;
+		dma-coherent;
 	};
 	ethernet@8 {
 		compatible = "fsl,dpa-ethernet";
 		fsl,fman-mac = <&enet6>;
+		dma-coherent;
 	};
 };
 
-- 
2.17.0

